sinopia_client/src/api/LDPApi.js

```diff
@@ -148,13 +152,14 @@
       };
       var headerParams = {
         'Slug': opts['slug'],
+        'Link': opts['link'],
         'Content-Type': opts['contentType']
       };
       var formParams = {
       };
 
       var authNames = ['RemoteUser'];
-      var contentTypes = ['application/ld+json'];
+      var contentTypes = [];
```

Comment:
We can leave this blank because `'application/ld+json'` is the default or for some other reason?

Reply:
in most cases, we just want to send `application/ld+json`, but for non-RDF resource creation (profiles and resource templates), we want to send plain `application/json` (see https://github.com/LD4P/sinopia_server/pull/66/files#diff-9c98b1f96912899c02b40c5bb91a07dcR107).  this is another rough edge of the JS codegen tool re: default values:  if you specify a bunch of possible content types, it'll just pick the first json content type no matter what you try to specify.  see:
https://github.com/LD4P/sinopia_server/blob/master/sinopia_client/src/ApiClient.js#L418
https://github.com/LD4P/sinopia_server/blob/master/sinopia_client/src/ApiClient.js#L154-L170

whereas if you don't have a list of preferred content types in the swagger spec, e.g. you override the global list of acceptable content types to empty for a specific operation, then you can specify at the time of the call:
https://github.com/LD4P/sinopia_server/blob/master/sinopia_client/src/ApiClient.js#L424-L426
https://github.com/LD4P/sinopia_server/blob/group_and_resource_creation/swagger.yaml#L248
https://github.com/LD4P/sinopia_server/blob/group_and_resource_creation/swagger.yaml#L273 (could probably use a clearer description in light of this discussion)

override in example call here: https://github.com/LD4P/sinopia_server/blob/group_and_resource_creation/sinopia_client/test/api/LDPApi.spec.js#L107

so the particular line that is the subject of this comment is essentially just the translation of the override for `createResource` in the auto-gen SDK code.